### PR TITLE
Adding support for kafka rest client

### PR DESCRIPTION
**Pull request by fx19880617**
If no proxyPort been set, kafka rest client won't be initialized.

### CODE REVIEW

**Comment by Raynos** [comment]
lgtm.

Want to add some integration tests ?

File: test/kafka.js [hunk 1]
```diff
@@ -35,3 +37,62 @@ test('kafka logging', function (assert) {
...
+        } else if (req.method === 'POST') {
+            assert.ok(req.headers.timestamp);
+            assert.equal(req.url, '/topics/rt-foobarx');
+            var body = '';
+            req.on('data', function (data) {
+                body += data;
+            });
+            req.on('end', function () {
+                assert.ok(body.indexOf('info') !== -1);
+                assert.ok(body.indexOf('writing to kafka') !== -1);
```
Comment (Raynos) [inline test/kafka.js]: just do a `res.end()` here.

File: test/kafka.js [hunk 2]
```diff
@@ -35,3 +37,62 @@ test('kafka logging', function (assert) {
...
+            });
+        }
+    }).listen(restProxyPort);
+    restProxyServer.addListener( "connection", function( socket ) {
```
Comment (Raynos) [inline test/kafka.js]: no need for this hack.

File: test/kafka.js [hunk 3]
```diff
@@ -35,3 +37,62 @@ test('kafka logging', function (assert) {
...
+                leafPort: server.port,
+                proxyHost: 'localhost',
+                proxyPort: restProxyPort
+            })
+        }
+    });
+
+    logger.info('writing to kafka');
+    setTimeout(function() {
+        logger.destroy();
```
Comment (Raynos) [inline test/kafka.js]: assert.equal(count, 1)

File: test/kafka.js [hunk 4]
```diff
@@ -35,3 +37,62 @@ test('kafka logging', function (assert) {
...
+
+test('kafka logging with rest client', function(assert) {
+    var server = KafkaServer(function onMessage(msg) {
+        assert.equal(msg.topic, 'rt-foobarx');
+        var obj = msg.messages[0].payload;
+        assert.equal(obj.level, 'info');
+        assert.ok(obj.msg.indexOf('writing to kafka') !== -1);
+        server.close();
+    });
+
```
Comment (Raynos) [inline test/kafka.js]: var count = 0

File: test/kafka.js [hunk 5]
```diff
@@ -35,3 +37,62 @@ test('kafka logging', function (assert) {
...
+            res.end(JSON.stringify(messages));
+        } else if (req.method === 'POST') {
+            assert.ok(req.headers.timestamp);
+            assert.equal(req.url, '/topics/rt-foobarx');
+            var body = '';
+            req.on('data', function (data) {
+                body += data;
+            });
+            req.on('end', function () {
+                assert.ok(body.indexOf('info') !== -1);
```
Comment (Raynos) [inline test/kafka.js]: count++

File: test/kafka.js (hunk 1 above)
Comment (fx19880617) [inline test/kafka.js]: Added

File: test/kafka.js (hunk 2 above)
Comment (fx19880617) [inline test/kafka.js]: removed

File: test/kafka.js (hunk 3 above)
Comment (fx19880617) [inline test/kafka.js]: done

File: test/kafka.js (hunk 4 above)
Comment (fx19880617) [inline test/kafka.js]: done

File: test/kafka.js (hunk 5 above)
Comment (fx19880617) [inline test/kafka.js]: done

File: test/kafka.js [hunk 6]
```diff
@@ -35,3 +37,60 @@ test('kafka logging', function (assert) {
...
+                assert.ok(body.indexOf('writing to kafka') !== -1);
+            });
+            count++
```
Comment (Raynos) [inline test/kafka.js]: definitely want to increment count++ after the assert.ok inside the callback

**Comment by Raynos** [comment]
lgtm.

**Comment by Raynos** [comment]
Fails lint :) Can you fix linting issues.

**Pull request merged**